src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -239,7 +240,12 @@ def insertBlockRules(self):
                           ignore_availability=True, meta=self.metaData)
             rseName = "%s_Test" % item['pnn'] if self.testRSEs else item['pnn']
             # DATASET = replicates all files in the same block to the same RSE
-            resp = self.rucio.createReplicationRule(item['blockname'],
+            if self.isT0agent:
+                kwargs.update(self.datasetRuleParams)
```

Comment:
I'd suggest to remove this if/else. By default it will be set to an empty dictionary, so it's harmless to any system unless the agent configuration is modified.

Reply:
Ok, I just wanted to be extra cautious not trying to break any other system.